lib/src/native/realm_core.dart

```diff
@@ -39,63 +58,78 @@ class _RealmCore {
 
   String get libraryVersion => _realmLib.realm_get_library_version().cast<Utf8>().toDartString();
 
+  LastError? getLastError([Allocator? allocator]) {
+    if (allocator != null) {
+      final error = _realmLib.realm_get_last_error();
+
+      if (error == nullptr) {
+        return null;
+      }
+
+      String? message;
+      if (error.ref.message != nullptr) {
+        message = error.ref.message.cast<Utf8>().toDartString();
+      }
+      _realmLib.realm_release_last_error(error);
+
+      return LastError(error.ref.error, message);
+    }
+
+    return using((Arena arena) {
+      return getLastError(arena);
+    });
+  }
+
   //TODO: Use Finalizers, when available, instead of native WeakHandles https://github.com/dart-lang/language/issues/1847
-  //The proposal: https://github.com/dart-lang/language/blob/master/working/1847%20-%20FinalizationRegistry/proposal.md
   SchemaHandle createSchema(List<SchemaObject> schema) {
     return using((Arena arena) {
       final classCount = schema.length;
 
-      final schemaClasses = arena.allocate<realm_class_info_t>(classCount * sizeOf<realm_class_info_t>());
-      final schemaProperties = arena.allocate<Pointer<realm_property_info_t>>(classCount * sizeOf<Pointer<realm_property_info_t>>());
+      final schemaClasses = arena<realm_class_info_t>(classCount);
+      final schemaProperties = arena<Pointer<realm_property_info_t>>(classCount);
 
       for (var i = 0; i < classCount; i++) {
         final schemaObject = schema.elementAt(i);
         final classInfo = schemaClasses.elementAt(i).ref;
 
-        classInfo.name = schemaObject.name.toNativeUtf8().cast();
-        classInfo.primary_key = "".toNativeUtf8().cast();
+        classInfo.name = schemaObject.name.toUtf8Ptr(arena);
+        classInfo.primary_key = "".toUtf8Ptr(arena);
```

Comment:
I assume the empty string is intended here as a temporary, until we are further (just checking)

Reply:
yes we need to update it when we do the primary key support. Perhaps I should add a TODO there, to get analyzer help